packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -199,14 +215,33 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
       port: 80
     };
 
+    if (props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS && props.certificate !== undefined) {
```

Comment:
Seems simpler to assign the default value for `protocol` first and then check it's equal to `HTTPS`?